src/Traits/ReflectionClassLikeTrait.php

```diff
@@ -442,7 +443,7 @@ public function getParentClass()
             $extendsNode = $hasExtends ? $this->classLikeNode->$extendsField : null;
             if ($extendsNode instanceof FullyQualified) {
                 $extendsName = $extendsNode->toString();
-                $parentClass = class_exists($extendsName, false) ? new parent($extendsName) : new static($extendsName);
+                $parentClass = class_exists($extendsName, false) ? new BaseReflectionClass($extendsName) : new ReflectionClass($extendsName);
```

Comment:
I was only really suggesting adding a  `protected function createReflectionObjectForClass()` containing only that one trinary expression line. Seems like a pretty minor change to me.